jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java

```diff
@@ -737,20 +745,26 @@ protected void doUpdate(String id, SessionData data)
                 statement.setLong(5, data.getExpiry());
                 statement.setLong(6, data.getMaxInactiveMs());
 
-                try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
-                     ObjectOutputStream oos = new ObjectOutputStream(baos))
+                if(!data.getAllAttributes().isEmpty())
                 {
-                    SessionData.serializeAttributes(data, oos);
-                    byte[] bytes = baos.toByteArray();
-                    try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes))
+                    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                         ObjectOutputStream oos = new ObjectOutputStream(baos))
                     {
-                        statement.setBinaryStream(7, bais, bytes.length);//attribute map as blob
-                        statement.executeUpdate();
-
-                        if (LOG.isDebugEnabled())
-                            LOG.debug("Updated session " + data);
+                        SessionData.serializeAttributes(data, oos);
+                        byte[] bytes = baos.toByteArray();
+                        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes))
+                        {
+                            statement.setBinaryStream( 7, bais, bytes.length );//attribute map as blob
+                        }
                     }
                 }
+                else
+                {
+                    statement.setBinaryStream( 7, EMPTY, 0);
```

Comment:
Does this deserialize correctly? Serializing an empty map would be a different result than an empty byte string.

Reply:
just an empty map but good point I will add a test for that.